scheduler/pkg/kafka/pipeline/utils.go

```diff
@@ -46,12 +48,26 @@ func createResourceNameFromHeader(header string) (string, bool, error) {
 		"Bad or missing header %s %s", resources.SeldonModelHeader, header)
 }
 
+func addRequestIdToKafkaHeadersIfMissing(headers []kafka.Header, requestId string) []kafka.Header {
+	for _, kafkaHeader := range headers {
+		if kafkaHeader.Key == util.RequestIdHeader { //already exists
+			return headers
+		}
+	}
+	return append(headers, kafka.Header{
+		Key:   util.RequestIdHeader,
+		Value: []byte(requestId),
+	})
+}
+
+// We ensure the Kafka headers are lower case as http headers may have been canonical uppercased
 func convertHttpHeadersToKafkaHeaders(httpHeaders http.Header) []kafka.Header {
 	var kafkaHeaders []kafka.Header
 	for k, vals := range httpHeaders {
-		if strings.HasPrefix(strings.ToLower(k), resources.ExternalHeaderPrefix) {
+		key := strings.ToLower(k)
```

Comment:
:upside_down_face: Minor point, but we only need to convert the form of the key if we're going to propagate it

Reply:
We do propogate it and x-request-id (lowercase) is used amongst others,. When these are converted back to http later if needed they will revert to case when added.